packages/@lwc/engine-core/src/framework/shadow-migration-mode.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { logWarnOnce } from '../shared/logger';
+
+let globalStylesheet: CSSStyleSheet | undefined;
+
+function isStyleElement(elm: Element): elm is HTMLStyleElement {
+    return elm.tagName === 'STYLE';
+}
+
+async function fetchStylesheet(elm: HTMLStyleElement | HTMLLinkElement) {
+    if (isStyleElement(elm)) {
+        return elm.textContent;
+    } else {
+        // <link>
+        const { href } = elm;
+        try {
+            return await (await fetch(href)).text();
+        } catch (err) {
+            logWarnOnce(`Ignoring cross-origin stylesheet in migrate mode: ${href}`);
+            // ignore errors with cross-origin stylesheets - nothing we can do for those
+            return '';
+        }
+    }
+}
+
+function initGlobalStylesheet() {
+    const stylesheet = new CSSStyleSheet();
+    const elmsToPromises = new Map();
+    let lastSeenLength = 0;
+
+    const copyToGlobalStylesheet = () => {
+        const elms = document.head.querySelectorAll(
+            'style:not([data-rendered-by-lwc]),link[rel="stylesheet"]'
+        );
+        if (elms.length === lastSeenLength) {
```

Comment:
I'm not sure if this is a concern but if someone changed the `href`` on the link for example:

`<link rel=stylesheet src=app.css>` -> `<link rel=stylesheet src=app2.css>`

This would miss that update.

Reply:
Yep, good point! I doubt we will need to handle that case, though. I think in most cases people are just adding/removing link tags. But I can add a comment about this.

Reply:
9ae57b06a